### ISSUE TITLE
API key security

### ISSUE THREAD

**Issue by yaelberger-commits** [cds-snc/notification-planning#1402]
## Description

Stronger language when an API key is created that it cannot be shared anywhere publicly.
More details in the API documentation, linked from the Terms of Use, on how to keep API keys secure. 

As a developer and Notify API user, I need information documented in obvious places so that I understand how to keep my API keys secure and so I can follow the Terms of Use for GC Notify.

WHY are we building?
For better API key security and so our users are able to follow our Terms of Use so we don't have security issues.
WHAT are we building?
Updated documentation in the UI and in the API documentation
VALUE created by our solution
Better comprehension from developers of how to keep their API keys safe and fewer security issues or incidents related to API keys

## Documentation and Artifacts
From UK Notify's new Terms of Use:
NotifyUK has some new terms of use, including nice stuff around api keys
API keys
If you use the Notify API to send messages, you must protect your API keys from unauthorised access or disclosure. This includes encrypting the keys at rest and in transit. You should also rotate your keys:
at least once every year
when someone with access to a key leaves your team
whenever you suspect that a key may have been compromised
Do not hard-code your API keys.
We recommend that you use a cloud service provider’s key management service to keep your API keys secure. For more information, read the NCSC guidance.

## Acceptance Criteria
Given a developer is using the GC Notify API, when they read the Terms of Use and the API documentation while onboarding, then they understand the steps to keep their API keys safe

Given a developer is creating an API key in GC Notify, then when they take that step, they are presented in the UI with further details reminding them how to keep their API keys secure.
- [x] Update API content
- [x] Update TOU content
- [x] Fix [Heading sometimes appears in EN in FR TOU#1784](https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/gh/cds-snc/notification-planning/1784)
- [x] Create new version of Security statement with the new content and link to previous version
- [x] Update google doc of Notify policy docs once content is approved and can link to new API content
- [x] Cypress UI tests if needed.
- [x] Generate appropriate log messages so that executions of this feature can be tracked
- [ ] Can misuse of this feature cause harm? If yes, create an alert
- [ ] Update the status of related findings, insights, and hypotheses on the Research Airtable
- [ ] Once change/fix/feature is implemented, link relevant Airtable records to design artifacts (Figma)

* A11y
* Bilingualism
* Privacy considerations
* Security controls in place
* Measuring success and metrics

## Related [Research Airtable](https://airtable.com/appWwAmHwDLtpIyko/tbl38n7ofWYBuezFc/viwQUX3tUxbMxYS7C?blocks=hide) records
- UK Notify updated their Terms of Use and we have some information from past support tickets/issues

## QA Steps
- [ ] Tested in a realistic production scenario

## GC Articles Publish checklist
- [  ] Update google doc of Notify policy docs once content is approved
- [ ]  Supporting feature is released and unflagged
- [ ]  Create or update slug
- [ ]  Page content and slug is translated
- [ ]  Document links to and from the page
- [ ]  Add page slug and title to [inventory](https://docs.google.com/document/d/1N6B4fSwoM40m-pDLI70BvLfL7O0n04ndyAXYaIwNbNw/edit)
- [ ]  If new page, add WAF rule
- [ ]  Create new page or update existing page content
- [ ]  Create or update translation to the page
- [ ]  Link translations by choosing an assigned translation
- [ ]  If new page, save as draft
- [ ]  If existing page, update the page
- [ ]  Post page in Slack for last minute reviews
- [ ]  On staging: Clear the cache through GC Notify's admin panel
- [ ]  On staging: Check links, images and language toggle
- [ ]  Give a heads up to the app-team channel once ready to publish
- [ ]  On staging and production: Clear the cache through GC Notify's admin panel
- [ ]  Clear Cloudfront cache in GCA as needed (Andrew and Phil are GCA admins)

**Comment by yaelberger-commits** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Notify-Planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/1402?planning-poker) with Zenhub @jzbahrai @andrewleith @mtoutloff @whabanks @YedidaZalik

**Comment by YedidaZalik** [comment]
Prepared table of Notify security content/locations comparing to UK content
Final column is To do and To ask 
Need to get answers on the To ask part

**Comment by YedidaZalik** [comment]
Worked on content and checking if there's a Canadian reference we can use for https://www.ncsc.gov.uk/collection/developers-collection/principles

Content is mostly done and here https://docs.google.com/document/d/1P0VupegAe2fm540A4z3P3Kw2VPpQ3eDEHgMYipauTjA/edit?tab=t.0

Still working on banner for API key creation here https://www.figma.com/design/jMvTot5NT4VwWl7YYXqPay/Integrating-the-API-%7C-Int%C3%A9grer-l'API?node-id=1647-307&t=wMOYyBJNHOYq23wQ-0

**Comment by yaelberger-commits** [comment]
@YedidaZalik to continue with this one today if time allows

**Comment by YedidaZalik** [comment]
Implemented feedback from Jumana and Melissa
Will send to localization once get decision on one outstanding question

**Comment by YedidaZalik** [comment]
Jumana approved the suggestion, prepped content for localization and sent to Marie-Sophie
https://docs.google.com/document/d/1HnklNaZkG3pA6M9Qjve7dEi2OfGGTO0PgUG76DTR7pw/edit?tab=t.0#heading=h.etnz92lonhiu

**Comment by YedidaZalik** [comment]
Localization is back, embedded in the doc.

**Comment by YedidaZalik** [comment]
Updated GC Articles pages
Changes showing to Security statement, including archived version
TOU header change not showing so Andrew will look into it

**Comment by YedidaZalik** [comment]
Jumana reviewed Security stmt changes
TOU heading changes need dev work 
Jumana will look at it

**Comment by YedidaZalik** [comment]
Jumana has fixed this, ready for review
Yedida will look

**Comment by YedidaZalik** [comment]
Updated FR Security statement with a few changes I'd missed

**Comment by yaelberger-commits** [comment]
@YedidaZalik and @jzbahrai please check off acceptance criteria above as you go. Is this work now completed?

**Comment by andrewleith** [comment]
TOU heading has been updated. @YedidaZalik, is this card all done now?

**Comment by YedidaZalik** [comment]
No, in the last bullet we need to hyperlink the words "Secure your API keys" to the steps they need to follow in the API documentation
Jumana said I cannot do it, it needs to be done by a dev 
I'm happy to do it if that's not the case!

**Comment by andrewleith** [comment]
PR to update this content: https://github.com/cds-snc/notification-admin/pull/2129

**Comment by andrewleith** [comment]
@YedidaZalik here is the link on the PR review environment if you want to sign off on the change: https://vgpquenxlfl6aqmg6bzkhwj7ym0uakat.lambda-url.ca-central-1.on.aws/terms

**Comment by YedidaZalik** [comment]
@andrewleith I'm sorry to be a pain
We want to link to the specific section of the Security notice

In English it's https://vgpquenxlfl6aqmg6bzkhwj7ym0uakat.lambda-url.ca-central-1.on.aws/security#:~:text=Always%20follow%20these%20steps%3A

In French https://notification.canada.ca/securite#:~:text=Suivez%20toujours%20les%20%C3%A9tapes%20suivantes%C2%A0%3A
